Add initial functions (#1)
urfave/cli-validation: README.md

# cli-validation

> Library to help with flag validation of urfave/cli(current for v3 only). This ensures that urfave/cli codebase remains compact and clean of optional features

Reviewer: An example of what flag validation is would be nice. Is it like type checking, range checking? If `cli` still handles some basic checks, maybe it worth mention which.

Author: @abitrolly Updated README.md to reflect what this library does. Let me know if that clarifies.

Reviewer: That's a very good explanation. Thanks.

> Library which contains help validators for use with the flag validation feature of urfave/cli(current for v3 only). Validation in urfave/cli is run only after the value has been converted from a string to the flag value type. So these validators need to focus only on validating the value, whether it be within a certain range or picked from a fixed number of values. For slice/map flags validation cannot be used to ensure that the underlying structure is of a certain length(s) since the final slice/map len can be known only after all the flags have been parsed. That kind of validation is a candidate for a Flag Action since those actions are run after all flags have been parsed

Reviewer: Reworded a bit.

```suggestion
Library with additional validators for flag validation feature of `urfave/cli` (current for v3 only). Validation in `urfave/cli` is run only after the value has been converted from a string to the flag value type. So these validators need to focus only on validating the value, whether it be within a certain range or picked from a fixed number of values. Validation cannot be used to check slice/map length, because len can be known only after all the flags have been parsed. That kind of validation is a candidate for a Flag Action since those actions are run after all flags have been parsed
```
